lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +247,94 @@
         NativeModule.require;
       fn(this.exports, requireFn, this, process);
 
+      if (config.experimentalModules && !NativeModule.isInternal(this.id)) {
+        this.exportKeys = Object.keys(this.exports);
+
+        const update = (property, value) => {
+          if (this.reflect !== undefined &&
+              ObjectHasOwnProperty.call(this.reflect.exports, property))
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+        const methodUnwrapMap = new WeakMap();
+
+        // mitigates https://crbug.com/v8/5773 where calling
+        // a native method with a proxy receiver will throw.
+        const wrap = (target, name, value) => {
+          if (typeof value !== 'function' || !isNative(value)) {
+            return value;
+          }
+
+          if (methodWrapMap.has(value))
+            return methodWrapMap.get(value);
+
+          const p = new Proxy(value, {
+            apply: (t, thisArg, args) => {
+              if (thisArg === proxy) {
+                thisArg = target;
+              }
+              return ReflectApply(t, thisArg, args);
+            },
+            __proto__: null,
+          });
+
+          methodWrapMap.set(value, p);
+          methodUnwrapMap.set(p, value);
+
+          return p;
+        };
+
+        const proxy = new Proxy(this.exports, {
+          set: (target, prop, value, receiver) => {
+            if (typeof value === 'function')
+              value = methodUnwrapMap.get(value) || value;
+            if (ReflectSet(target, prop, value, receiver)) {
+              update(prop, ReflectGet(target, prop, receiver));
+              return true;
+            }
+            return false;
+          },
+          defineProperty: (target, prop, descriptor) => {
+            if (ObjectHasOwnProperty.call(descriptor, 'value')) {
+              const { value } = descriptor;
+              if (typeof value === 'function')
+                descriptor.value = methodUnwrapMap.get(value) || value;
+            }
+            if (ReflectDefineProperty(target, prop, descriptor)) {
+              update(prop, ReflectGet(target, prop));
+              return true;
+            }
+            return false;
```

Comment:
this line gets hit when you do:

```mjs
import events from 'events';
Object.defineProperty(events, 'defaultMaxListeners', {value: 333})

/*
TypeError: 'defineProperty' on proxy: trap returned falsish for property 'defaultMaxListeners'
    at Function.defineProperty (<anonymous>)
    at file:///Users/bfarias/Documents/node/t.mjs:2:8
    at ModuleJob.run (internal/modules/esm/module_job.js:106:14)
*/
```

But it should probably report this as an error for redefining like it does right now:

```mjs
> Object.defineProperty(events, 'defaultMaxListeners', {value: 333})
TypeError: Cannot redefine property: defaultMaxListeners
    at Function.defineProperty (<anonymous>)
```

Reply:
@devsnek @bmeck By changing the `defineProperty` trap from
```js
if (ReflectDefineProperty(target, prop, descriptor)) {
  update(prop, ReflectGet(target, prop));
  return true;
}
return false;
```
to
```js
ObjectDefineProperty(target, prop, descriptor)
update(prop, ReflectGet(target, prop));
return true;
```
we get the expected error:
```
TypeError: Cannot redefine property: defaultMaxListeners
    at Function.defineProperty (<anonymous>)
```